substrate/frame/society/src/lib.rs

```diff
@@ -1488,13 +1503,58 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		let rotation_period = voting_period + claim_period;
 		let now = T::BlockNumberProvider::current_block_number();
 		let phase = now % rotation_period;
-		if phase < voting_period {
+		if now > Self::next_intake_at() {
+			Period::IntakeDelay { elapsed: now - Self::next_intake_at() }
+		} else if phase < voting_period {
 			Period::Voting { elapsed: phase, more: voting_period - phase }
 		} else {
 			Period::Claim { elapsed: phase - voting_period, more: rotation_period - phase }
 		}
 	}
 
+	/// Next intake (candidate/membership) rotation scheduled with [Config::BlockNumberProvider].
+	pub fn next_intake_at() -> BlockNumberFor<T, I> {
+		NextIntakeAt::<T, I>::get().unwrap_or({
+			// executed once.
+			let now = T::BlockNumberProvider::current_block_number();
+			let prev_block = now.saturating_sub(BlockNumberFor::<T, I>::one());
+			let rotation_period = T::VotingPeriod::get().saturating_add(T::ClaimPeriod::get());
+			let elapsed = prev_block % rotation_period;
+			prev_block + (rotation_period - elapsed)
```

Comment:
Note: the subtraction here cannot underflow since we do the % first. Otherwise it could underflow it would result in a very big value since its using u32.